wsclient/src/test/java/org/hpccsystems/ws/client/utils/ConnectionTest.java

```diff
@@ -0,0 +1,110 @@
+package org.hpccsystems.ws.client.utils;
+
+/*******************************************************************************
+ *     HPCC SYSTEMS software Copyright (C) 2022 HPCC Systems®.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *******************************************************************************/
+
+import static org.junit.Assert.*;
+
+import java.net.MalformedURLException;
+
+import org.junit.Test;
+
+public class ConnectionTest

```

Comment:
@jchambers-ln not sure how you encountered this issue, if you provide details, we can include junit coverage.

Reply:
I saw that even though a https was passed in that the isHttps wasn't set right

Reply:
ok, in that case the new junit tests should provide enough coverage.